### ISSUE TITLE
Access to custom deployment from the outside

### ISSUE THREAD

**Issue by yurkoff-mv** [kubeflow/kubeflow#6658]
/kind question

**Question:**
How can I createa custom deployment and access it over the network from the outside?
I have a cluster with the Kubeflow installed. I have my own custom deployment (not InferenceService, nor anything else is suitable). I create Service and VirtualService for my Deployment and send a request to it. In response, I get only the Dex authorization page.

```
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: echo-server-v1
  labels:
    version: v1
spec:
  selector:
    matchLabels:
      app: echo-server
      version: v1
  replicas: 2
  template:
    metadata:
      labels:
        app: echo-server
        version: v1
    spec:
      containers:
      - name: echo-server
        image: k8s.gcr.io/echoserver:1.5
---
apiVersion: v1
kind: Service
metadata:
  name: echo-server-v1
  labels:
    version: v1
spec:
  type: ClusterIP
  ports:
  - name: echo-port
    port: 80
    targetPort: 8080
  selector:
    app: echo-server
    version: v1
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: echo-external-services
spec:
  hosts:
  - "*"
  gateways:
  - istio-ingressgateway
  - kubeflow-gateway
  - knative-local-gateway
  http:
  - match:
    - uri:
        exact: /echo
    route:
    - destination:
        host: echo-server-v1.kubeflow.svc.cluster.local
        port:
          number: 80
```
```
!kubectl get svc -n istio-system -l istio=ingressgateway

NAME                   TYPE       CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                                      AGE
istio-ingressgateway   NodePort   10.152.183.95   <none>        15021:31650/TCP,80:30264/TCP,443:32643/TCP,31400:32107/TCP,15443:30681/TCP   18d
```
```
curl http://127.0.0.1:30264
<a href="/dex/auth?client_id=kubeflow-oidc-authservice&amp;redirect_uri=%2Flogin%2Foidc&amp;response_type=code&amp;scope=profile+email+groups+openid&amp;state=MTY2Mzc0OTE4NHxFd3dBRUd4Q1ZURXhXRVpKYTBKNlVYaDJOV1U9fPv6qwe9rpzvviU4afOAmOON4vgoOypOHDcvyY0YL7VV">Found</a>.
```

When i pass authorization in Dashboard, it gives out the following
![image](https://user-images.githubusercontent.com/82467993/191462383-6a457539-1d3d-4a81-b198-91bacb4dae34.png)

**Environment:**

- Istio Version: 1.9
- Knative Version: 0.22.1
- KFServing Version: 0.7.0
- Kubeflow version: 1.5
- Kubernetes version: (use kubectl version): Microk8s 1.20
- OS (e.g. from /etc/os-release): Ubuntu 20.04

**Comment by yurkoff-mv** [comment]
I solved the problem. Here are the files I was able to do this with.
[echo-server.zip](https://github.com/kubeflow/kubeflow/files/9747883/echo-server.zip)
And here is a diagram of the interaction between Kubernetes and KubeFlow components.
![image](https://user-images.githubusercontent.com/82467993/194922792-d3703590-002b-40f0-afc7-6b852681e526.png)
The echo server is deployed in a separate namespace with a separate Gateway .
